### ISSUE TITLE
X11 and Wayland permissions

### ISSUE THREAD

**Issue by lucasmz-dev** [zocker-160/SyncThingy#38]
The flatpak contains X11 and Wayland permissions and I notice that STy does not work without them, which in theory on what it needs to do is a bit weird, since it is just Syncthing and a tray icon. Having access to these AFAIK means access to the clipboard, holes like screen-copy on KDE, etc. It would be nice if Syncthingy could work without these (and by extension also not need IPC since AFAIK that's only used for X11 perf) 

Basically, something similar to how Jellyfin Server works: https://github.com/flathub/org.jellyfin.JellyfinServer
(It does not contain a tray icon, but... yk)

**Comment by zocker-160** [comment]
The only permission I think I can remove is the `--share=ipc` one, if that is what you mean. I admit that I took over the manifest from another C++ / Qt application and they all had that permission.

I do need the `wayland` and the `fallback-x11` permission, otherwise the settings dialog would not work.
`dri` is needed, otherwise it will fall back to software rendering, the filesystem one should be self explanatory.
Both talk names are required to make notifications and the tray icon work.

**Comment by lucasmz-dev** [comment]
> otherwise the settings dialog would not work.

Oh... I did not notice that was a thing. This explains that.

**Comment by zocker-160** [comment]
If you right click the icon, you can select "Settings", it gives you a few options to configure.

For Steam Deck users this menu is very important, because it also gives you the ability to install it as a system service. 

If you are concerned about the security, you can disable those permissions using Flatseal, the sync process will still work (I just tried), you won't have a tray icon though.
